src/GNDStk/Node/src/sort.hpp

```diff
@@ -0,0 +1,90 @@
+
+// -----------------------------------------------------------------------------
+// Helpers
+// Private, for use by sort(). Could lead to surprises for users if not used
+// in the proper sort() context. In particular, less() assumes that any child
+// nodes it inspects were sorted already; if they weren't, then transitivity
+// in sort()'s strict weak ordering requirement would in general be violated,
+// leading to undefined behavior. I suppose less() could chuck this assumption
+// and instead take it upon itself to sort its child nodes, recursively, before
+// proceeding with the relevant comparisons. This would render the full sort()
+// as s-l-o-w as molasses, and would trash up the principle that less() should
+// compare nodes, not mess with them. As it stands, our full node::sort() does
+// things in the right order, so that less() works as it should in that context.
+// -----------------------------------------------------------------------------
+
+private:
+
+// less
+bool less(const Node &b) const
```

Comment:
There are any number of ways that we could choose to define `<` for a Node. This one is fine for me. I'm not sure of a specific instance where we would want to sort, but I may be mistaken.

Reply:
The "node sort" capability is basically there as a debugging aid. Imagine, for instance, that one test of the JSON read/write is as follows: for each of the 5000+ GNDS XMLs that are currently available, (1) read the XML, (2) write as JSON, (3) read the JSON, (4) compare the internal contents from (1) with those from (3) to see that they're identical. That's fine and good, except that nlohmann JSON sorts key/value pairs lexicographically by key. So we can't do the comparison easily unless we sort first in some predictable way. Roughly speaking, the new sort() replaces my old and much maligned normalize(). :-p Just think about it as strictly for debugging purposes.